src/storage_manifests.py

```diff
@@ -145,6 +155,51 @@ def evaluate(self) -> Optional[str]:
                 return f"Storage manifests waiting for definition of {prop}"
         return None
 
+    def is_ready(self, obj: HashableResource, cond: AnyCondition) -> Optional[bool]:
+        """Determine if the resource is ready."""
+        if not super().is_ready(obj, cond):
+            object_events = collect_events(self.client, obj.resource)
+            if obj.kind in ["Deployment", "DaemonSet"]:
+                involved_pods = self.client.list(
+                    Pod, namespace=obj.namespace, labels={"app": obj.name}
+                )
+                object_events += [
+                    event for pod in involved_pods for event in collect_events(self.client, pod)
+                ]
+            for event in sorted(object_events, key=by_localtime):
+                log.info(
+                    "Event %s/%s %s msg=%s",
+                    event.involvedObject.kind,
+                    event.involvedObject.name,
+                    event.lastTimestamp
+                    and event.lastTimestamp.astimezone()
+                    or "Date not recorded",
```

Comment:
localize the timestamp -- but sometimes events don't have an associated timestamp so we'll just say "Date not recorded" on those